modules/fee-market/src/s2s/payment.rs

```diff
@@ -216,12 +244,56 @@ where
 	rewards_book
 }
 
-/// Do slash for absent assigned relayers
-pub(crate) fn do_slash<T: Config<I>, I: 'static>(
+/// Calculate the reward for the order which has been confirmed in time.
+pub(crate) fn cal_rewards_before_deadline<T: Config<I>, I: 'static>(
+	slot_relayer_id: &T::AccountId,
+	message_relayer_id: &T::AccountId,
+	confirm_relayer_id: &T::AccountId,
+	total_reward: BalanceOf<T, I>,
+	base_fee: BalanceOf<T, I>,
```

Comment:
What is base fee?

The final fee for relayer?

Reply:
`base_fee` is the third assigned relayer's fee of the order, Used `reward_item.to_treasury = Some(total_reward.saturating_sub(base_fee));`.